addons/website_sale/controllers/main.py

```diff
@@ -576,16 +576,16 @@ def checkout_form_validate(self, mode, all_form_values, data):
         error = dict()
         error_message = []
 
-        if data.get('partner_id'):
+        if mode[1] == 'billing' and data.get('partner_id'):
```

Comment:
Checking the mode is the correct fix, but it should be done specifically for the `can_edit_vat` check to avoid disabling the internal user check for internal users who change their billing address.
```suggestion
        if data.get('partner_id'):
```